File: 1817-CalculateMoneyInLeetcodeBank/1817-CalculateMoneyInLeetcodeBank.py
# Last updated: 2/11/2026, 11:08:19 AM
class Solution:
    def totalMoney(self, n: int) -> int:
        arr = [0,1,2,3,4,5,6]
        res = 0
        index = 0

        for i in range(n):
            if (index >6):
                index = 0
            arr[index]+=1
            res+=arr[index]
            index+=1

        return res

        

File: 1817-CalculateMoneyInLeetcodeBank/1817-CalculateMoneyInLeetcodeBank.cpp
// Last updated: 2/11/2026, 11:08:17 AM
#include <vector>
using namespace std;

class Solution {
public:
    int totalMoney(int n) {
        vector<int> arr {0,1,2,3,4,5,6};
        int index =0;
        int res = 0;

        for (int i =0 ; i< n; ++i){
            if (index > 6){
                index = 0;
            }

            arr.at(index)++;
            res+= arr.at(index);
            index++;

        }
        return res;
    }

};